arch/arm/adler32_neon.c

```diff
@@ -171,12 +303,33 @@ Z_INTERNAL uint32_t adler32_neon(uint32_t adler, const uint8_t *buf, size_t len)
     /* If memory is not SIMD aligned, do scalar sums to an aligned
      * offset, provided that doing so doesn't completely eliminate
      * SIMD operation. Aligned loads are still faster on ARM, even
-     * though there's no explicit aligned load instruction */
-    unsigned int align_offset = ((uintptr_t)buf & 15);
-    unsigned int align_adj = (align_offset) ? 16 - align_offset : 0;
+     * though there's no explicit aligned load instruction. Note:
+     * on Android and iOS, their ABIs specify stricter alignment
+     * requirements for the 2,3,4x register ld1 variants. Clang for
+     * these platforms emits an alignment hint in the instruction for exactly
+     * 256 bits. Several ARM SIPs have small penalties for cacheline
+     * crossing loads as well (so really 512 bits is the optimal alignment
+     * of the buffer). 32 bytes should strike a balance, though. Clang and
+     * GCC on linux will not emit this hint in the encoded instruction and
```

Comment:
Just a nitpick, but "linux" as a generic term kind of implies also any other operating system based on Linux... For clarity, it should be always capitalized.